keylime/src/tpm.rs

```diff
@@ -225,6 +305,345 @@ impl Context {
+                    PublicEccParametersBuilder::new()
+                        .with_symmetric(SymmetricDefinitionObject::Null)
+                        .with_ecc_scheme(EccScheme::EcDsa(HashScheme::new(
+                            name_alg,
+                        )))
+                        .with_curve(curve_id)
+                        .with_key_derivation_function_scheme(
+                            KeyDerivationFunctionScheme::Null,
+                        )
+                        .with_is_signing_key(obj_attrs.sign_encrypt())
+                        .with_is_decryption_key(obj_attrs.decrypt())
+                        .with_restricted(obj_attrs.restricted())
+                        .build()?,
+                )
+                .with_ecc_unique_identifier(EccPoint::new(
+                    EccParameter::try_from(&UNIQUE_IDEVID[0..6])?,
+                    EccParameter::try_from(&UNIQUE_IDEVID[0..6])?,
+                )),
+            // Defaulting to RSA on null
+            AsymmetricAlgorithm::Null => PublicBuilder::new()
+                .with_public_algorithm(PublicAlgorithm::Rsa)
+                .with_name_hashing_algorithm(HashingAlgorithm::Sha256)
+                .with_object_attributes(obj_attrs)
+                .with_auth_policy(Digest::try_from(
+                    IDEVID_AUTH_POLICY_SHA256[0..32].to_vec(),
+                )?)
+                .with_rsa_parameters(
+                    PublicRsaParametersBuilder::new()
+                        .with_symmetric(SymmetricDefinitionObject::Null)
+                        .with_scheme(RsaScheme::Null)
+                        .with_key_bits(RsaKeyBits::Rsa2048)
+                        .with_exponent(RsaExponent::default())
+                        .with_is_signing_key(obj_attrs.sign_encrypt())
+                        .with_is_decryption_key(obj_attrs.decrypt())
+                        .with_restricted(obj_attrs.decrypt())
+                        .build()?,
+                )
+                .with_rsa_unique_identifier(PublicKeyRsa::try_from(
+                    &UNIQUE_IDEVID[0..6],
+                )?),
+        };
+
+        Ok(IDevIDPublic {
+            public: key_builder.build().unwrap(),
+        })
+    }
+
+    // Creates IAK
+    pub fn create_iak(
+        &mut self,
+        asym_alg: AsymmetricAlgorithm,
+        name_alg: HashingAlgorithm,
+    ) -> Result<IAKResult> {
+        let key_pub =
+            Self::create_iak_public_from_default_template(asym_alg, name_alg);
+
+        let pcr_selection_list = PcrSelectionListBuilder::new()
+            .with_selection(
+                HashingAlgorithm::Sha256,
+                &[
+                    PcrSlot::Slot0,
+                    PcrSlot::Slot1,
+                    PcrSlot::Slot2,
+                    PcrSlot::Slot3,
+                    PcrSlot::Slot4,
+                    PcrSlot::Slot5,
+                    PcrSlot::Slot6,
+                    PcrSlot::Slot7,
```

Comment:
Why do you pass this specific set of PCRs here?

Reply:
I suppose the state of these PCRs indicates the state of the machine when the key was created? Are you going to check anything with these PCRs? I am wondering how their state is going to be used.